src/bloqade/pyqrack/task.py

```diff
@@ -36,3 +37,12 @@ def state_vector(self) -> list[complex]:
         """Returns the state vector of the simulator."""
         self.run()
         return self.state.sim_reg.out_ket()
+
+    def qubits(self) -> list[PyQrackQubit]:
+        """Returns the qubits in the simulator."""
+
+        N = self.state.sim_reg.num_qubits()
```

Comment:
If you don't run the task before calling this method, you'll get a not very helpful error message: ´AttributeError: 'StackMemory' object has no attribute 'sim_reg'´.

I'd suggest you wrap the access to ´.sim_reg´ in a ´try - except´ (gotta love idiomatic python) and run the task if you catch an ´AttributeError´.

Also, it would be nice to have this method available directly on the simulator, similar to `.state_vector`.